addons/api/addon/services/store.js

```diff
@@ -0,0 +1,21 @@
+import Store, { CacheHandler } from '@ember-data/store';
+import RequestManager from '@ember-data/request';
+import { LegacyNetworkHandler } from '@ember-data/legacy-compat';
+import { inject as service } from '@ember/service';
+import IndexedDbHandler from 'api/handlers/indexed-db-handler';
+
+export default class extends Store {
+  // =services
+  @service indexedDb;
+
+  requestManager = new RequestManager();
+
+  constructor(args) {
+    super(args);
+
+    // Bind `this` so the handler can have access to the indexedDb service
+    IndexedDbHandler.request = IndexedDbHandler.request.bind(this);
```

Comment:
Not a fan of needing to do this but couldn't think of a better way to handle this. Open to suggestions